packages/jni-swig-stub/src/main/jni/realm_api_helpers.cpp

```diff
@@ -135,27 +135,30 @@ register_object_notification_cb(realm_object_t *object, jobject callback) {
 }
 
 void register_login_cb(realm_app_t *app, realm_app_credentials_t *credentials, jobject callback) {
-    auto jenv = get_env();
-    static jclass notification_class = jenv->FindClass("io/realm/internal/interop/CinteropCallback");
-    static jmethodID on_success_method = jenv->GetMethodID(notification_class, "onSuccess",
-                                                           "(Lio/realm/internal/interop/NativePointer;)V");
-    static jmethodID on_error_method = jenv->GetMethodID(notification_class, "onError", "(Ljava/lang/Throwable;)V");
-
     realm_app_log_in_with_credentials(
             app,
             credentials,
             // FIXME Refactor into generic handling of network requests, like
             //  https://github.com/realm/realm-java/blob/master/realm/realm-library/src/main/cpp/io_realm_internal_objectstore_OsApp.cpp#L192
             [](void *userdata, realm_user_t *user, realm_error_t *error) {
+                // TODO Investigate why method look ups in callbacks seems to need to be non-static
+                //  to avoid throwing
+                //  "JNI DETECTED ERROR IN APPLICATION: use of deleted local reference 0x55"
```

Comment:
This seems like it would hit us in a lot of places? Maybe create an issue, so we can reference that and also have more visibility around it.

Reply:
Isolated the issue to be that `FindClass` returns a local reference while `GetMethodID` returns a stable ID that can be statically referenced. We should make a global cache of these as global refs. Created #460 to track this.